c_src/trans_nif.c

```diff
@@ -89,11 +89,13 @@ derivate(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
   if (count < 1) // can't be empty
     return enif_make_badarg(env);
 
-  if (! (target = (decimal*) enif_make_new_binary(env, (count - 1) * sizeof(decimal), &r)))
+  if (! (target = (decimal*) enif_make_new_binary(env, count * sizeof(decimal), &r)))
     return enif_make_badarg(env); // TODO return propper error
 
+  if (count > 0)
+    target[0] = (decimal) {.coefficient = 0, .exponent = 0, .confidence = 0};
```

Comment:
The question is if we use interpolation after or before the operation, doing it after will give a more 'natural' feeling, changes are that the derivate before is close to the derivate now, instead of 0

Think about network throughput for example mesured by total packages:

if we  have the measurements:

10000 10010 100020 100030 100040 

the derivate (w/o filling in the 1st value) would be:
10 10 10 10

with your approach:
0 10 10 10 10

with my approach:

10 10 10 10 10

now this of cause is a very baiesed example :P

But for greater values this would be even more obvious, you'd end up having large jumps to 0